### ISSUE TITLE
ValueError: 'shape' elements cannot be negative

### ISSUE THREAD

**Issue by kosmylo** [cvxgrp/cvxpy#1072]
**Describe the bug**
Instead of solving the problem, I am getting: 
ValueError: 'shape' elements cannot be negative

**To Reproduce**
```python
charging_rate_opt = cp.Variable((total_num_EVs, opt_horizon))
charging_demand_opt = cp.Variable((total_num_EVs, opt_horizon))

revenue = 0
constraints = []

for t in range(opt_horizon-1):
    
    for i in range(total_num_EVs):
    
        constraints += [ charging_rate_opt[i, t] >= 0.0, charging_rate_opt[i, t] <= max_charging_rate, charging_demand_opt[i, t] >= 0.0]
        
        revenue -= price[t] * charging_rate_opt[i, t]
    
        if connectivity_matrix[i, t] == 2:
                        
            constraints += [ charging_demand_opt[i, t] == charging_demand_init[i] ]
            
            if 1 <= parking_time_init[i] <= 2:
                
                revenue += 1.2 * price[t] * charging_demand_init[i]
            
            elif 3 <= parking_time_init[i] <= 6:
                
                revenue += 1.1 * price[t] * charging_demand_init[i]
                
            else:
                
                revenue += 1.05 * price[t] * charging_demand_init[i]
            
            constraints += [ charging_demand_opt[i, t+1] == charging_demand_opt[i, t] - charging_rate_opt[i, t] ]
            
            if connectivity_matrix[i, t+1] == 0:
            
                revenue -= 1.5 * price[t] * charging_demand_opt[i, t]
        
        if connectivity_matrix[i, t] == 1:  
            
            constraints += [ charging_demand_opt[i, t+1] == charging_demand_opt[i, t] - charging_rate_opt[i, t] ]
            
            if connectivity_matrix[i, t+1] == 0:
            
                revenue -= 1.5 * price[t] * charging_demand_opt[i, t]
                
        if connectivity_matrix[i, t] == 0:
            
            constraints += [ charging_rate_opt[i, t] == 0.0 ]
                      
prob = cp.Problem(cp.Maximize(revenue), constraints)
prob.solve(solver=cp.MOSEK, verbose = True)
```

**Version**
 - OS: Windows 10
 - CVXPY Version: 1.1.1

I noticed in some other issues that some solutions provided, but no solution worked.

**Comment by BlankCanvasStudio** [comment]
Hey. I don't know if you are still having this problem or what solver the problem is in but I have a solution to the problem if you are using a conic solver. I have my own public repository where I fixed the overflow issue you are facing. If you download my version of cvxpy instead of this one it should work just fine (at least it does for me). You just have to cast certain values from int32 to int64 and it works just fine. I'm very new to github and I don't know how to create a pull request here since I can't edit this repo so I created my own. The rest of my repo is exactly the same, I only fixed the overflow issue in the conic solver and canonInterface. I'm using the same OS and CVXPY version so if it doesn't work let me know and I can try to fix the issue for your particular case.

**Comment by BlankCanvasStudio** [comment]
So I read too fast and didn't notice you were using the MOSEK solver. I think I can fix the problem for your case but the variables you use aren't given values in your code and I don't want to just take shots in the dark until I get the error so can you share your full code so I can try and get you a solution? My current code doesn't fix your problem as I haven't fixed the overflow error in the MOSEK solver but if you send me a full version of your code I can run a debugger through their code and (hopefully) fix the issue!

**Comment by SteveDiamond** [comment]
@BlankCanvasStudio That's great you've fixed this issue! To make a PR, you fork the cvxpy repo and work from there.

**Comment by rileyjmurray** [comment]
I'm closing this issue, since it should be resolved by PR #1074. @kosmylo please let us know if the latest master-branch version of CVXPY does not fix this problem for you.
